File: src/src/main/java/com/mewebstudio/translatable/repository/CategoryTranslationRepository.java
package com.mewebstudio.translatable.repository;

import com.mewebstudio.springboot.jpa.translatable.JpaTranslationRepository;
import com.mewebstudio.translatable.entity.Category;
import com.mewebstudio.translatable.entity.CategoryTranslation;

public interface CategoryTranslationRepository extends JpaTranslationRepository<CategoryTranslation, String, Category> {
    boolean existsByLocaleAndName(String locale, String name);

    boolean existsByLocaleAndNameAndIdIsNot(String locale, String name, String id);
}

File: src/main/kotlin/com/mewebstudio/translatable/repository/CategoryTranslationRepository.kt
package com.mewebstudio.translatable.repository

import com.mewebstudio.springboot.jpa.translatable.kotlin.JpaTranslationRepository
import com.mewebstudio.translatable.entity.Category
import com.mewebstudio.translatable.entity.CategoryTranslation

interface CategoryTranslationRepository : JpaTranslationRepository<CategoryTranslation, String, Category> {
    fun existsByLocaleAndName(locale: String, name: String): Boolean

    fun existsByLocaleAndNameAndIdIsNot(locale: String, name: String, id: String): Boolean
}
